File: src/main/java/by/architecture_map/belarus/entity/Image.java
package by.architecture_map.belarus.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Inheritance;
import jakarta.persistence.InheritanceType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

import lombok.Getter;
import lombok.Setter;

import java.util.Objects;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@Getter
@Setter
public class Image extends BaseEntity {

    private String url;
    private Boolean show;

    @ManyToOne
    @JoinColumn(name = "source_id")
    private Source source;

    private String author;

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof Image)) return false;

        Image image = (Image) other;
        return Objects.equals(getId(), image.getId()) &&
                Objects.equals(url, image.url) &&
                Objects.equals(show, image.show) &&
                Objects.equals(source, image.source) &&
                Objects.equals(author, image.author);
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId(), url, source, author);
    }
}

File: src/main/kotlin/by/architecture_map/belarus/entity/Image.kt
package by.architecture_map.belarus.entity

import jakarta.persistence.Entity
import jakarta.persistence.Inheritance
import jakarta.persistence.InheritanceType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
open class Image(

    open var url: String? = null,
    open var show: Boolean? = null,
    @ManyToOne
    @JoinColumn(name = "source_id")
    open var source: Source? = null,
    open var author: String? = null

) : BaseEntity() {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Image) return false

        if (id != other.id) return false
        if (url != other.url) return false
        if (show != other.show) return false
        if (source != other.source) return false
        if (author != other.author) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + url.hashCode()
        result = 31 * result + source.hashCode()
        result = 31 * result + author.hashCode()
        return result
    }
}
